Sources/OktaLogger/Analytics/AnalyticsProviderProtocol.swift

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2020-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import Foundation
+import OktaLogger
+
+/**
+ Protocol: Client needs to implement this protocol and provide the information about provider
+ */
+@objc
+public protocol AnalyticsProviderProtocol: AnyObject {
+
+    /// provider name, eg: Firebase, AppCenter etc.
+    var name: String { get set }
+    /// logger to post logs on it's destination
+    var logger: OktaLoggerProtocol { get set }
```

Comment:
Why logger is mandatory for AnalyticsProvider? Do really need it to be part of AnalyticsProvider?

Reply:
Yes this can be optional  as App might choose not to pass a Logger instance to the analytics framework. App might only want use Analytics functionality